node/src/reactor/main_reactor/catch_up.rs

```diff
@@ -316,10 +310,17 @@ impl MainReactor {
         block_hash: BlockHash,
     ) -> CatchUpInstruction {
         let sync_leap_identifier = SyncLeapIdentifier::sync_to_tip(block_hash);
-        let peers_to_ask = self.net.fully_connected_peers_random(
-            rng,
-            self.chainspec.core_config.simultaneous_peer_requests as usize,
-        );
+        let peers_to_ask_count = self.chainspec.core_config.simultaneous_peer_requests as usize;
+        let peers_to_ask = self
+            .net
+            .fully_connected_peers_random(rng, peers_to_ask_count);
+        if peers_to_ask.len() < peers_to_ask_count {
+            warn!(
+                "CatchUp: attmepting to Leap with less connected peers ({}) than required ({})",
```

Comment:
```suggestion
                "CatchUp: attempting to Leap with less connected peers ({}) than wanted ({})",
```

Reply:
Fixed in https://github.com/casper-network/casper-node/pull/3507/commits/8352a002fa1f86ed2a3ae82ec8440c3a00961e4e